graylog2-server/src/main/java/org/graylog2/rest/models/system/cluster/responses/NodeSummary.java

```diff
@@ -56,6 +59,10 @@ public boolean isMaster() {
     @JsonProperty
     public abstract String hostname();
 
+    @JsonProperty
+    @Nullable
+    public abstract DataNodeStatus datanodeStatus();
```

Comment:
What about creating a DataNodeSummary class that would provide all these datanode-specific methods and the nodeservice would return the correct implementation based on the type / database entry info?